### PR TITLE
Adds VPN Control Center widget and Siri Commands.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/1203108348835387/1208485618883319/f

DRK PR: https://github.com/duckduckgo/DesignResourcesKit/pull/9

## Description

Adds a VPN control widget for the control center and lock screen.
Adds Siri commands to control the VPN with voice.

### General

1. Go to Settings > VPN > VPN Settings > Shortcuts section
2. Read the education screens and follow their indications to validate them
3. Test the new CC Widget and Siri commands as per instructions.

### Control Center Widget

1. Go to the control center dashboard
4. Add a widget
5. Search for "VPN", and you should see our widget to toggle the VPN on/off.  Add it.
6. You can resize the widget in the dashboard before saving.
7. Turn the VPN on/off, make sure it works.
8. Go to the lock screen, press on the background for a few seconds.
9. Customize your lock screen.
10. Remove one of the two circle-widgets in the bottom of the lock screen, and add the VPN one.
11. Make sure it works well.

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/NetworkProtectionStatusViewModel.swift
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtectionStatusViewModel.swift:471]: Just makes sure to also refresh the new control widget.

File: DuckDuckGo/NetworkProtectionVPNSettingsView.swift
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtectionVPNSettingsView.swift:39]: This was moved to shortcuts view alongside other new education views.

File: DuckDuckGo/NetworkProtectionVPNSettingsView.swift
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtectionVPNSettingsView.swift:168]: Need to localize.

File: DuckDuckGo/NetworkProtectionWidgetRefreshModel.swift
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtectionWidgetRefreshModel.swift:39]: Just making sure we also reload the control center widget.

File: DuckDuckGo/Siri/SiriBubbleView.swift
Comment (diegoreymendez) [inline DuckDuckGo/Siri/SiriBubbleView.swift:23]: Chat bubble coming up in the new Siri education view.

File: DuckDuckGo/VPNAutoShortcuts.swift
Comment (diegoreymendez) [inline DuckDuckGo/VPNAutoShortcuts.swift:28]: These are the commands the user can call with voice, or in Shortcuts.app

File: Core/PixelEvent.swift
Comment (diegoreymendez) [inline Core/PixelEvent.swift:1788]: Pixel Details: https://app.asana.com/0/1206580121312550/1208989200355298/f

File: DuckDuckGo/DuckDuckGo.entitlements
Comment (diegoreymendez) [inline DuckDuckGo/DuckDuckGo.entitlements:13]: Support for Siri commands.

File: DuckDuckGo/Info.plist
Comment (diegoreymendez) [inline DuckDuckGo/Info.plist]: Xcode deciding to move everything around.  I'll look at whether I can roll this back, shortly.

File: DuckDuckGo/NetworkProtectionStatusView.swift
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtectionStatusView.swift:68]: This guard was missing, but the VPN education sheet should be only available in iOS 17+.

File: DuckDuckGo/VPNAutoShortcuts.swift
Comment (diegoreymendez) [inline DuckDuckGo/VPNAutoShortcuts.swift:25]: We may have other shortcuts with different iOS requirements, so I added the iOS version to the name here.  Shouldn't make a huge difference though.

File: DuckDuckGo/VPNIntents.swift
Comment (diegoreymendez) [inline DuckDuckGo/VPNIntents.swift:110]: I moved these waiting loops to the controller's `start` method.

File: DuckDuckGo/VPNIntents.swift
Comment (diegoreymendez) [inline DuckDuckGo/VPNIntents.swift:98]: Should be moved to the widget intents.

File: DuckDuckGo/VPNSnoozeActivityAttributes.swift
Comment (diegoreymendez) [inline DuckDuckGo/VPNSnoozeActivityAttributes.swift:22]: This wasn't necessary here.

File: Widgets/WidgetVPNIntents.swift
Comment (diegoreymendez) [inline Widgets/WidgetVPNIntents.swift:113]: Adding all this so I can fire a cancelled pixel.

File: Widgets/WidgetVPNIntents.swift
Comment (diegoreymendez) [inline Widgets/WidgetVPNIntents.swift:49]: I moved most of the shared logic into this controller and intents are now more specific to where they're triggered from, so we can fire specific pixels depending on source.

File: LocalPackages/VPNiOS/Sources/VPNWidgetSupport/VPNWidgetTunnelController.swift
Comment (samsymons) [inline LocalPackages/VPNiOS/Sources/VPNWidgetSupport/VPNWidgetTunnelController.swift:75]: Per our chat, let's add a 30 second cap on this to be safe.

File: LocalPackages/VPNiOS/Package.swift
Comment (samsymons) [inline LocalPackages/VPNiOS/Package.swift:1]: Noting here that we're using Swift 6; I think this is fine in this case, but we should probably align as a team on how we want to approach this in the new year.

File: DuckDuckGo/Widgets/Education/ControlCenterWidgetEducationView.swift
Comment (samsymons) [inline DuckDuckGo/Widgets/Education/ControlCenterWidgetEducationView.swift]: Are we able to uncomment or delete this one?

File: DuckDuckGo/VPNShortcutIntents.swift
Comment (samsymons) [inline DuckDuckGo/VPNShortcutIntents.swift]: We can clean these up since the `ForegroundContinuableIntent` type's availability will us this for free.

File: DuckDuckGo.xcodeproj/project.pbxproj
Comment (samsymons) [inline DuckDuckGo.xcodeproj/project.pbxproj]: This looks to be empty, we can remove it from the build phases.

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (samsymons) [inline DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: We'll want to swap this out with a tagged release before merging

File: DuckDuckGo/Siri/SiriEducationView.swift
Comment (samsymons) [inline DuckDuckGo/Siri/SiriEducationView.swift]: I think the strings in this file will have to be localized

File: DuckDuckGo/VPNAutoShortcuts.swift
Comment (samsymons) [reply DuckDuckGo/VPNAutoShortcuts.swift:25]: I love this, thank you!

File: DuckDuckGo/VPNAutoShortcuts.swift
Comment (samsymons) [inline DuckDuckGo/VPNAutoShortcuts.swift]: I suspect this should be:

```suggestion
                        "Stop the VPN connection with \(.applicationName)"
```

File: DuckDuckGo/VPNAutoShortcuts.swift
Comment (samsymons) [inline DuckDuckGo/VPNAutoShortcuts.swift:31]: How should localization be handled for Siri Shortcuts?

File: DuckDuckGo/VPNShortcutIntents.swift
Comment (samsymons) [inline DuckDuckGo/VPNShortcutIntents.swift]: Should these also be handled in a localized way?

File: DuckDuckGo/Widgets/Education/WidgetEducationView.swift
Comment (samsymons) [inline DuckDuckGo/Widgets/Education/WidgetEducationView.swift:39]: I tested this on old OS versions, looks good!

File: DuckDuckGo.xcodeproj/project.pbxproj
Comment (diegoreymendez) [reply DuckDuckGo.xcodeproj/project.pbxproj]: Removed.

File: LocalPackages/VPNiOS/Package.swift
Comment (diegoreymendez) [reply LocalPackages/VPNiOS/Package.swift:1]: Got it.  Marking as resolved as there's no change required.

File: DuckDuckGo/Widgets/Education/ControlCenterWidgetEducationView.swift
Comment (diegoreymendez) [reply DuckDuckGo/Widgets/Education/ControlCenterWidgetEducationView.swift]: Re-enabled.

File: DuckDuckGo/VPNShortcutIntents.swift
Comment (diegoreymendez) [reply DuckDuckGo/VPNShortcutIntents.swift]: Removed.

File: DuckDuckGo/VPNAutoShortcuts.swift
Comment (diegoreymendez) [reply DuckDuckGo/VPNAutoShortcuts.swift:31]: This [isn't exactly trivial](https://sowenjub.me/writes/localizing-app-shortcuts-with-app-intents/) so I'm opening [a follow-up task to look into this](https://app.asana.com/0/1206580121312550/1209026082513680/f).

File: DuckDuckGo/VPNShortcutIntents.swift
Comment (diegoreymendez) [reply DuckDuckGo/VPNShortcutIntents.swift]: Localized.

File: DuckDuckGo/Siri/SiriEducationView.swift
Comment (diegoreymendez) [reply DuckDuckGo/Siri/SiriEducationView.swift]: Localized.

File: DuckDuckGo/VPNShortcutIntents.swift
Comment (diegoreymendez) [reply DuckDuckGo/VPNShortcutIntents.swift]: I had to roll these back, because it's not easy to map our existing localization mechanism to `LocalizedStringResource`.  The compiler complains about the passed string not being a literal.

That said these are localized string resources so we'll need to figure out how to map these to our regular mechanism.

FWIW though I think that's exposed is the shortcut name, and not the intent name.

Comment (diegoreymendez) [reply DuckDuckGo/VPNShortcutIntents.swift]: Added a second bullet [here](https://app.asana.com/0/1206580121312550/1209026082513680/f).

File: LocalPackages/VPNiOS/Sources/VPNWidgetSupport/VPNWidgetTunnelController.swift
Comment (diegoreymendez) [reply LocalPackages/VPNiOS/Sources/VPNWidgetSupport/VPNWidgetTunnelController.swift:75]: Capped.

**Review by samsymons — APPROVED** [review]
LGTM after the recent changes, with the acknowledgement that we'll revisit in the new year to sort out localization. But, since this change is gated behind a `DEBUG || ALPHA` check, it's fine to merge. Thanks!

**Pull request merged**
